Core/src/org/sleuthkit/autopsy/mainui/datamodel/AnalysisResultDAO.java

```diff
@@ -855,6 +855,13 @@ void clearCaches() {
 
     @Override
     Set<? extends DAOEvent> processEvent(PropertyChangeEvent evt) {
+        
+        if(evt.getPropertyName().equals(Case.Events.ANALYSIS_RESULT_DELETED.toString())) {
+            this.analysisResultCache.invalidateAll();
```

Comment:
@kellykelly3 I think this is probably incorrect. AnalysisResultDAO has 3 caches - analysisResultCache, setHitCache, and keywordHitCache. We either somehow need to check what kind of analysis result it is and invalidate the appropriate cache (e.g. KW hits), or invalidate all of them.

Reply:
@eugene7646 You are correct, which is why the deleted keyword was not disappearing.  I am going to invalidate them all. You asked that I look at the keyword and tree issues, which I am currently working on.